kafka/pykafka/consumer.py

```diff
@@ -29,15 +133,63 @@ def topic(self):
     def partitions(self):
         return self._partitions
 
+    @property
+    def fetch_message_max_bytes(self):
+        return self._fetch_message_max_bytes
+
     def __iter__(self):
         while True:
-            yield self.consume()
+            message = self.consume(timeout=self._socket_timeout_ms)
+            if not message and self._consumer_timed_out():
+                raise StopIteration
+            yield message
```

Comment:
Oh, this isn't what I expected.  We'll need to document that iterating this isn't exactly an infinite iterator, but I think I like this solution better.